arcade/arcade/worker/mcp/sse.py

```diff
@@ -0,0 +1,447 @@
+                    )
+                    raise
+
+            logger.debug(f"secrets: {secrets}")
+            logger.debug(f"Calling tool {name} with arguments: {arguments}")
+            logger.debug(f"Tool context: {tool_context.model_dump_json()}")
+
+            # Execute the tool
+            result = await self.executor.run(
+                tool.tool,
+                tool.definition,
+                tool.input_model,
+                tool.output_model,
+                context=tool_context,
+                **arguments,
+            )
+
+            # Convert the result to MCP content format
+            if result.value is not None:
+                return convert_to_mcp_content(result.value)
+            elif result.error is not None:
+                error_message = result.error.message
+                developer_message = result.error.developer_message or ""
+                traceback = result.error.traceback_info or ""
+                logger.error(f"Error calling tool {name}: {error_message}")
+                if developer_message or traceback:
+                    logger.debug(f"Developer details: {developer_message}\n{traceback}")
+                return [{"type": "text", "text": f"Error calling tool {name}: {error_message}"}]
+            else:
+                logger.error(f"Error calling tool {name}: Unexpected result format - {result}")
+                return [
+                    {"type": "text", "text": f"Error calling tool {name}: Unexpected result format"}
+                ]
+        except Exception as e:
+            logger.error(f"Error calling tool {name}: {e}", exc_info=True)
+            return [{"type": "text", "text": f"Error calling tool {name}: {e!s}"}]
+
+    def _get_auth_requirement(self, tool: MaterializedTool) -> AuthRequirement:
+        auth_requirement = None
+        if tool.definition.requirements and tool.definition.requirements.authorization:
+            requirement = tool.definition.requirements.authorization
+            if requirement.oauth2:
+                auth_requirement = AuthRequirement(
+                    provider_id=requirement.provider_id,
+                    provider_type=requirement.provider_type,
+                    oauth2=AuthRequirementOauth2(
+                        scopes=requirement.oauth2.scopes,
+                    ),
+                )
+            else:
+                auth_requirement = AuthRequirement(
+                    provider_id=requirement.provider_id,
+                    provider_type=requirement.provider_type,
+                )
+        return auth_requirement
+
+    async def _check_authorization(self, tool: MaterializedTool) -> AuthorizationResponse:
+        """Check if the tool is authorized to be called."""
+        arcade_client = AsyncArcade(api_key=config.api.key)
+        try:
+            user_id = config.user.email if config.user.email else self._get_connection_id()
```

Comment:
For discussion: connection IDs are not permanent (to my knowledge), so this could result in unexpected behavior (the user constantly being asked to re-authorize)

Reply:
that was the idea since we can't guarentee otherwise without worker local storage right?